base/server/cms/src/com/netscape/cms/profile/common/EnrollProfile.java

```diff
@@ -549,8 +553,21 @@ public void submit(IAuthToken token, IRequest request)
             CMS.debug(e);
         }
 
-        if (token == null){
-            CMS.debug(method + " auth token is null; agent manual approval required;");
+        /*
+         * this is where we decide whether to let agent do manual approval
+         *  or not
+	 * If auth.instance_id is not set, then request automatically goes
+	 * into queue for agent approval.
+         * If auth.explicitApprovalRequired is true, then the request goes into
+         * queue for agent approval even though auth and authz succeed.
+         */
+         ((token == null) || (explicitApprovalRequired == true)){
```

Comment:
The logic makes sense, but is this code missing an `if` statement? There is no CI in PKI 10.5 branch so the code is not compiled/tested.

Reply:
Good catch.  I'm not sure what happened, as I did test the patch.  I worry I might have pick up from the wrong tree, so I fixed that and tested it again to work.